openslides/core/static/js/core/site.js

```diff
@@ -1315,15 +1321,37 @@ angular.module('OpenSlidesApp.core.site', [
             $http.post('/rest/core/projector/1/control_view/', {"action": action, "direction": direction});
         };
         $scope.editCurrentSlide = function () {
-            $.each(Projector.get(1).elements, function(key, value) {
-                if (value.name == 'agenda/list-of-speakers') {
+            angular.forEach(Projector.get(1).elements, function(element) {
+                if (element.name == 'agenda/list-of-speakers') {
                     $state.go('agenda.item.detail', {id: value.id});
                 } else if (
-                    value.name != 'agenda/item-list' &&
-                    value.name != 'core/clock' &&
-                    value.name != 'core/countdown' &&
-                    value.name != 'core/message' ) {
-                    $state.go(value.name.replace('/', '.')+'.detail.update', {id: value.id});
+                    element.name != 'agenda/item-list' &&
+                    element.name != 'core/clock' &&
+                    element.name != 'core/countdown' &&
+                    element.name != 'core/message' &&
+                    element.name != 'core/speakeroverlay') {
+                    $state.go(element.name.replace('/', '.')+'.detail.update', {id: element.id});
+                }
+            });
+        };
+        //*** List of speakers overlay on slide***
+        $scope.speakeroverlaytoggle = function () {
+            var data = {};
+            if ($scope.speakeroverlay.visible) {
+                data[$scope.speakeroverlay.uuid] = { "visible": false };
+            } else {
+                data[$scope.speakeroverlay.uuid] = { "visible": true };
+            }
+            $http.post('/rest/core/projector/1/update_elements/', data);
+        };
+        $scope.goToListofSpeakers = function() {
+            angular.forEach(Projector.get(1).elements, function(element) {
+                if (element.name == 'motions/motion') {
+                    $state.go('agenda.item.detail', {id: Motion.get(element.id).agenda_item_id});
```

Comment:
You may use DS.get(element.name, element.id).agenda_item_id. But before you run $state.go it would be more readable to finish the loop. E. g. you can use _.find or something similar to check if a related slide is on the projector. Then you use DS.get(result.name, result.id) ...